components/script/script_thread.rs

```diff
@@ -2008,6 +2008,17 @@ impl ScriptThread {
             ScriptThreadMessage::EvaluateJavaScript(pipeline_id, evaluation_id, script) => {
                 self.handle_evaluate_javascript(pipeline_id, evaluation_id, script, can_gc);
             },
+            ScriptThreadMessage::SendImageKeysBatch(pipeline_id, image_keys) => {
+                if let Some(window) = self.documents.borrow().find_window(pipeline_id) {
+                    window
+                        .image_cache()
+                        .fill_key_cache_with_batch_of_keys(image_keys);
+                } else {
+                    error!(
+                        "Could not find window corresponding to an image cache to send image keys to"
```

Comment:
I would add the pipeline, and only `warn`, because the pipeline may have closed in the meantime.